src/Stratis.Bitcoin/P2P/PeerAddressManager.cs

```diff
@@ -91,22 +91,24 @@ public void SavePeers()
         }
 
         /// <inheritdoc/>
-        public void AddPeer(IPEndPoint endPoint, IPAddress source)
+        public PeerAddress AddPeer(IPEndPoint endPoint, IPAddress source)
         {
-            this.AddPeerWithoutCleanup(endPoint, source);
+            PeerAddress peerAddress = this.AddPeerWithoutCleanup(endPoint, source);
 
             this.EnsureMaxItemsPerSource(source);
+
+            return peerAddress;
         }
 
-        private void AddPeerWithoutCleanup(IPEndPoint endPoint, IPAddress source)
+        private PeerAddress AddPeerWithoutCleanup(IPEndPoint endPoint, IPAddress source)
         {
             if (!endPoint.Address.IsRoutable(true))
-                return;
+                return null;
```

Comment:
we should add a trace here on the reason why we return null

Reply:
I will add the trace.